js/util/actor.js

```diff
@@ -55,10 +61,16 @@ Actor.prototype.receive = function(message) {
     }
 };
 
-Actor.prototype.send = function(type, data, callback, buffers) {
+Actor.prototype.send = function(type, data, callback, buffers, targetMapId) {
     var id = callback ? this.mapId + ':' + this.callbackID++ : null;
     if (callback) this.callbacks[id] = callback;
-    this.postMessage({ mapId: this.mapId, type: type, id: String(id), data: data }, buffers);
+    this.postMessage({
+        targetMapId: targetMapId,
+        mapId: this.mapId,
```

Comment:
There are really two different things here:
 - For messages initiated from the main thread, `mapId` identifies the _sending_ map instance, which is really for the purpose of enabling the worker to track state related to different instances.
 - For messages initiated from the worker, `targetMapId` identifies the _target_ map instance.

It's true that since these two properties are used in totally disjoint cases, we could simply use `mapId` for both.... but following along your point above re: making the code better self-documenting, I think it's a bit clearer not to overload the same property for two different needs (especially since they're similar, but subtly different)